text/052-platform-release-cycle.md

```diff
@@ -0,0 +1,174 @@
+   (using commit hashes, not necessarily tags).
+   This step should take no more than 2 weeks.
+   - The RM **tags** VX+rc, but produces no binary installers, see the platform release.
+   - The RM writes an **OPAM package**, currently the policy is to upload it to
+     `core-dev`.
+   - Possibly, the RM builds a docker image, so that project maintainers can use it
+     in CI during (4).
+   
+   **No breaking change is allowed from now on, unless a severe problem is found.**
+2. The documentation is updated (eg. the Changes file) and eventual fixes
+   required by the platform are done. Ideally no other change is done.
+   This step should take no more than 2 weeks.
+   - The RM **tags** the .0 release.
+   - The RM writes an **OPAM package** for the main OPAM repository.
+3. In response to a severe bug report Coq devs make an hotfix in master which is
+   backported to vX.
+   - The RM **tags** the point release, possibly as soon
+    as the fix is available and merged.
+   - The RM writes an **OPAM package** for the main OPAM repository.
+
+## Platform
+
+When Coq VX+rc is tagged, the PRM branches vX
+
+4. Starting with the pins made by RM on Coq's CI all packages part of the
+   platform (or its core) are pinned (in accordance with upstreams, which are
+   notified about the ongoing process). Most, if not all, packages are in Coq's
+   CI or the Platform's CI so there should be no surprises.
+   This step should not take more than one month.
+   - The PRM **tags** VX.0+beta tag and publishes the **binary installers**
+   - The PRM may also provide a **docker image** with the entire platform prebuilt
+5. As users pick up the platform and find severe bugs in Coq, the platform picks
+   up point releases of Coq containing hotfixes and eventually extends packages
+   beyond the core set.
+
+### Platform versioning
+
+It is out of the scope of this CEP to chose a versioning schema for the
+platform. In the diagram above we append a .digit to the Coq version (one
+of the current proposals).
+
+## Synchronization points
+
+- The end of (1) starts the release cycle of the platform.
+- The end of (2) and (4) don't need to be done at the same time, but if they
+  are then the release cycle presented in this CEP roughly coincides with the
+  previous one. Normally (2) precedes (4), but if the doc is not ready yet
+  the platform can still release a beta.
+- Coq's VX.0 tag can be made as soon as the doc is clear. This will stress the
+  fact that upstreams can pin with no worries.
+
+## Announces
+
+- Coq tags are only announced to devs
+- Coq platform pins are communicated to upstream devs, which may tag/release.
+- Coq platform tags and packages **are announced to the community** (this is
+  when we party)
+
+## Coq's CI
+
+- must test **one configuration**: it cannot use one compiler version or
+  compiler flavor to test one project, and a different one to test another one,
+  since the platform can only have one
```

Comment:
This point is unclear and not actionable as is. It seems to be in contradiction with the flexibility left in the last two points below on not testing all platform pure Coq libraries and testing non-standard flavors. It indeed seems to be a good idea that CI tests several configurations (including edge versions of dependencies) so that problems are detected sooner rather than later.
Let's identify current (actual) issues in the Coq CI and fix them (for instance the point below relates to an actual issue).

Reply:
I could rephrase it is a more operational way.

For each project P included in the platform:
- if P has ML code, then it must be tested in Coq's CI using the platform configuration. It can also be tested using something else, but not exclusively using another configuration
- if P is pure .v code:
  - if it is a core component (has many users in the platform) the it must be tested in Coq's CI using the platform compiler. It can also be tested using something else, but not exclusively using a non platform compiler.
  - if it is not a core component, then it can be tested with any configuration in Coq's CI, or not tested at all
  - it must be tested by the platform's CI

A configuration is a version/flavor of the OCaml compiler and a set of configure options for Coq/env variables.

The decision of which is the platform configuration and the switch to a new one is subject to discussion between Coq and platform devs, and the switch should happen in a coordinated way.

W.r.t. the problems with Coq's CI, I think (hope) the only one left is about the configuration. IIRC many jobs disable native but I could not reproduce any failure using a standard 4.07.1 compiler on the same gitlab workers. So something tells me we tweak the configuration (I did not investigate yet).

Reply:
Is this text more clear? Should I write it this way in the CEP?

Reply:
Yes, much clearer. Yes, please update the CEP to be more like what you wrote in this comment.

Reply:
done